File: Advisor-Builder-Python/Yieldster-SDK/Models/ConditionalStep.py
from Models.Step import Step


class ConditionalStep(Step):
    def __init__(self):
        self.conditionalStatement = None
        self.trueExecuteStep = []
        self.falseExecuteStep = []
        self.type = "ConditionalStep"

    # def add_conditionalStatement(self, conditionalStatement):
    #     self.conditionalStatement = conditionalStatement
    #
    # def add_trueExecuteStep(self, trueExecuteStep):
    #     self.trueExecuteStep.append(trueExecuteStep)
    #
    # def add_falseExecuteStep(self, falseExecuteStep):
    #     self.falseExecuteStep.append(falseExecuteStep)

File: Advisor-Builder-JAVA/advisorBuilder/src/main/java/com/ddc/yieldster/jsonbuilder/model/ConditionalStep.java
package com.ddc.yieldster.jsonbuilder.model;

import lombok.Builder;
import lombok.Data;
import lombok.Value;

import java.util.List;

@Builder
@Value
@Data
public class ConditionalStep extends Step {
    String type = "ConditionalStep";
    String conditionalStatement;
    List<Step> trueExecuteStep;
    List<Step> falseExecuteStep;
}
